src/SQLite.cs

```diff
@@ -2501,7 +2483,57 @@ public TableMapping (Type type, CreateFlags createFlags = CreateFlags.None)
 			_insertColumns = Columns.Where (c => !c.IsAutoInc).ToArray ();
 			_insertOrReplaceColumns = Columns.ToArray ();
 		}
+
+		private IReadOnlyCollection<MemberInfo> GetPublicMembers(Type type)
+		{
+			if(type.Name.StartsWith("ValueTuple`"))
+				return GetFieldsFromValueTuple(type);
+
+			var members = new List<MemberInfo>();
+			var memberNames = new HashSet<string>();
+			var newMembers = new List<MemberInfo>();
+			do
+			{
+				var ti = type.GetTypeInfo();
+				newMembers.Clear();
 
+				newMembers.AddRange(
+					from p in ti.DeclaredProperties
+					where !memberNames.Contains(p.Name) &&
+						p.CanRead && p.CanWrite &&
+						(p.GetMethod != null) && (p.SetMethod != null) &&
+						(p.GetMethod.IsPublic && p.SetMethod.IsPublic) &&
+						(!p.GetMethod.IsStatic) && (!p.SetMethod.IsStatic)
+					select p);
+				
+				newMembers.AddRange(
+					from f in ti.DeclaredFields
+					where f.IsPublic && !f.IsStatic && !memberNames.Contains(f.Name)
+					select f);
```

Comment:
This is, in general good, except it's going to change the semantics of the library for current users. Right now, public fields are ignored implicitly on reference types. This will migrate databases to include them.

Instead, we should only add public fields for specific types. Probably, in this case, keep it constrained to `ValueTuples`. Not all reference types nor all value types. Eventually, we may want to open this to fields on value types but I will have to consider the full repercussions. 

So please check that this type is indeed a ValueTuple before accepting fields.

Reply:
Changed